src/IssueViz.Security/Taint/TaintStore.cs

```diff
@@ -62,19 +127,24 @@ public void Set(IEnumerable<IAnalysisIssueVisualization> issueVisualizations, An
                 throw new ArgumentNullException(nameof(issueVisualizations));
             }
 
-            this.analysisInformation = analysisInformation;
+            lock (Locker)
+            {
+                this.analysisInformation = analysisInformation;
+
+                var oldIssues = taintVulnerabilities;
+                taintVulnerabilities = issueVisualizations.ToList();
 
-            var oldIssues = taintVulnerabilities;
-            taintVulnerabilities = issueVisualizations.ToArray();
+                var removedIssues = oldIssues.Except(taintVulnerabilities).ToArray();
```

Comment:
This is comparing based on reference equality. Is this what we want?

Reply:
Hmm. Good question. I suspect then that we always had a bug here. This code is only called by Set(), when the monitor fetches new issues from the server. I think in reality it was always putting all old as removed and all new as added.

Reply:
I think the old code just assumed we were doing a complete replacement of issues every time i.e. it was just a batch update.